### PR TITLE
Returning an explicit session validation response instead of http status

### PR DESCRIPTION
**Pull request by dennisoelkers**
This helps us to prevent basic auth popups showing up for the request,
because of browsers intercepting 401s for non-CORS requests. Helps a lot
when the web interface is served over a single port using a relative
server URL like proposed and enabled in #2156.

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/rest/models/system/sessions/responses/SessionValidationResponse.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+package org.graylog2.rest.models.system.sessions.responses;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.auto.value.AutoValue;
+
+@AutoValue
+@JsonAutoDetect
+public abstract class SessionValidationResponse {
```
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/rest/models/system/sessions/responses/SessionValidationResponse.java]: Maybe it would be useful to add the user name of the user who created the session in the response.

Comment (dennisoelkers) [inline graylog2-server/src/main/java/org/graylog2/rest/models/system/sessions/responses/SessionValidationResponse.java]: Good idea. That would also help us because we don't need to keep it around in `localStorage` in the web interface. This would be done in a separate PR though.

**Comment by joschi** [comment]
LGTM.
